include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -103,11 +103,11 @@ class BandBlock {
   }
 
   void copyDiag(SizeType j, const ConstTileType& source) noexcept {
-    constexpr auto General = lapack::MatrixType::General;
-    constexpr auto Lower = lapack::MatrixType::Lower;
+    constexpr auto General = blas::Uplo::General;
```

Comment:
This changes were introduced to fix an annoying API choice made in lapackpp.
laset and lacpy use MatrixType instead of Uplo to specify upper/lower/general.
However both functions do not accept any of the other MatrixType specifiers and therefore:
- we have to always assert that the matrix type is always U/L/G
- we cannot use directly an Uplo argument

To solve this problem I defined the `lapack::laset` and `lapack::lacpy` wrappers with the first argument of type Uplo, and modified our tile operations to only use the Uplo argument.